### PR TITLE
O3-4250 (feat) Ability to add Cash Points and Payment modes via a UI

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary
Add UI functionality for creating billing settings during facility configuration
- Introduced a UI to display and creation of billing configuration
setting e.g. cash points and payment modes via tables
- Added functionality to create new billing related setting with zod validation
- Functionality to avoid duplicating existing cash points or payment modes

## Screenshots

https://github.com/user-attachments/assets/6fb4b044-4e89-4ce8-b02e-6d53392d6006

## Related Issue
https://openmrs.atlassian.net/browse/O3-4250

## Other
To do
- Add functionality to prevent deleting a cash point or payment mode related to 
and existing bill (support to be added to the backend first)

### CODE REVIEW

File: src/billable-services/cash-point/cash-point-configuration.component.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  TableCell,
+  Modal,
+  TextInput,
+  OverflowMenu,
+  OverflowMenuItem,
+  Dropdown,
+} from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { useTranslation } from 'react-i18next';
+import axios from 'axios';
```
Comment (ibacher) [inline src/billable-services/cash-point/cash-point-configuration.component.tsx:20]: Please, if at all possible, use openmrsFetch instead of axios.

Comment (ODORA0) [reply src/billable-services/cash-point/cash-point-configuration.component.tsx:20]: @ibacher Absolutely

**Review by CynthiaKamau — APPROVED** [review]

**Review by jabahum — APPROVED** [review]

File: src/billable-services/cash-point/cash-point-configuration.component.tsx [hunk 2]
```diff
@@ -114,21 +112,37 @@ const CashPointConfiguration: React.FC = () => {
     }
 
     try {
-      await axios.post(`${baseUrl}/billing/cashPoint`, {
-        name: data.name,
-        uuid: data.uuid,
-        location: { uuid: data.location },
+      const response = await openmrsFetch('/ws/rest/v1/billing/cashPoint', {
```
Comment (jabahum) [inline src/billable-services/cash-point/cash-point-configuration.component.tsx:115]: you could use `restBaseUrl` otherthan completing the entire url

File: src/billable-services/payyment-modes/payment-modes-config.component.tsx [hunk 3]
```diff
@@ -55,7 +53,7 @@ const PaymentModesConfig: React.FC = () => {
 
   const fetchPaymentModes = useCallback(async () => {
     try {
-      const response = await axios.get(`${baseUrl}/billing/paymentMode?v=full`);
+      const response = await openmrsFetch('/ws/rest/v1/billing/paymentMode?v=full');
```
Comment (jabahum) [inline src/billable-services/payyment-modes/payment-modes-config.component.tsx:56]: here too

File: src/billable-services/payyment-modes/payment-modes-config.component.tsx [hunk 4]
```diff
@@ -89,20 +87,37 @@ const PaymentModesConfig: React.FC = () => {
     }
 
     try {
-      await axios.post(`${baseUrl}/billing/paymentMode`, {
-        name: data.name,
-        description: data.description,
+      const response = await openmrsFetch('/ws/rest/v1/billing/paymentMode', {
```
Comment (jabahum) [inline src/billable-services/payyment-modes/payment-modes-config.component.tsx:90]: here too

File: src/billable-services/payyment-modes/payment-modes-config.component.tsx [hunk 5]
```diff
@@ -117,7 +132,9 @@ const PaymentModesConfig: React.FC = () => {
     if (!selectedPaymentMode) return;
 
     try {
-      await axios.delete(`${baseUrl}/billing/paymentMode/${selectedPaymentMode.uuid}`);
+      await openmrsFetch(`/ws/rest/v1/billing/paymentMode/${selectedPaymentMode.uuid}`, {
```
Comment (jabahum) [inline src/billable-services/payyment-modes/payment-modes-config.component.tsx:135]: here too

**Review by jabahum — COMMENTED** [review]
@ODORA0 just a few changes

File: src/billable-services/payyment-modes/payment-modes-config.component.tsx (hunk 3 above)
Comment (ODORA0) [reply src/billable-services/payyment-modes/payment-modes-config.component.tsx]: Done

File: src/billable-services/payyment-modes/payment-modes-config.component.tsx (hunk 4 above)
Comment (ODORA0) [reply src/billable-services/payyment-modes/payment-modes-config.component.tsx]: Done

**Pull request merged**